module5/TaskListApp/TaskListApp/Views/TaskDetailsView.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  TaskDetailsView.swift
+//  TaskListApp
+//
+//  Created by Peter Rice on 12/22/23.
+//
+
+import SwiftUI
+
+struct TaskDetailsView: View {
+  var tasksStore: TasksStore
+  let task: Task  
+  @State private var taskCompleted: Bool = false
+
+  var body: some View {
+    VStack {
+      Form {
+        Section(header:
+          Text("Task Title")
+        )
+        {
+          Text(task.title)
+            .bold()
+        }
+        Section(header: Text("Notes")) {
+          Text(task.notes)
+            .bold()
+        }
+        Section() {
+          Toggle(isOn: $taskCompleted) {
+            Text("Completed: ")
+              .bold()
+          }
+          .onChange(of: taskCompleted) {
```

Comment:
Is this the primary pattern for collecting and performing form input validation in SwiftUI? Ie you attach onChange() handlers to all of the inputs, and trigger validation logic on the form state from there?

Fwiw - i tried doing this w/ property closure on didSet but it didnt seem to trigger when the binding updated.

Reply:
`onChange` is one way, it will get called for each letter. You can also look at `focused(_:)` to validate when they tap away.

Reply:
Unless you really want mutliple lines, consider using a `TextField` as it has some built in options for formatting and validating.

Reply:
Ignore what I said about TextField. I thought I was looking at a different part of your code.